wodles/azure/tests/test_orm.py

```diff
@@ -0,0 +1,192 @@
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute
+# it and/or modify it under the terms of GPLv2
+
+
+import hashlib
+import json
+import os
+import pytest
+import sys
+
+from sqlalchemy import create_engine
+from sqlalchemy import inspect
+from sqlalchemy.exc import IntegrityError
+from unittest.mock import patch
+
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), '..'))  # noqa: E501
+with patch('sqlalchemy.create_engine', return_value=create_engine("sqlite://")):
+    import orm
+
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+test_last_dates_path = os.path.join(test_data_path, 'last_date_files')
+
+EXPECTED_TABLE_NAMES = ['graph', 'log_analytics', 'storage']
+
+
+@pytest.fixture(scope='function')
+def create_and_teardown_db():
+    orm.create_db()
+    yield
+    orm.Base.metadata.drop_all(orm.engine)
+
+
+@pytest.fixture(scope='function')
+def teardown_db():
+    yield
+    orm.Base.metadata.drop_all(orm.engine)
+
+
+def test_create_db(teardown_db):
+    # Check there is no tables available
+    inspector = inspect(orm.engine)
+    assert inspector.get_table_names() == []
+
+    # Create the tables
+    orm.create_db()
+
+    # Validate the tables and their structure
+    inspector = inspect(orm.engine)
+    table_names = inspector.get_table_names()
+    assert table_names == EXPECTED_TABLE_NAMES
+    for table in table_names:
+        assert [x['name'] for x in inspector.get_columns(table)] == ['md5', 'min_processed_date', 'max_processed_date']
+
+
+def test_add_get_row(create_and_teardown_db):
+    """Test the ORM is able to insert row into the different tables and retrieve them as a whole and individually as
+    expected."""
+    # Add several entries and validate them
+    for table in [orm.Graph, orm.LogAnalytics, orm.Storage]:
+        for id_ in range(3):
+            md5 = hashlib.md5(f'{table.__tablename__}{id_}'.encode()).hexdigest()
+            original_datetime = "2022-01-01T23:59:59.1234567Z"
+            row = table(md5=md5, min_processed_date=original_datetime, max_processed_date=original_datetime)
+            orm.add_row(row=row)
+            row = orm.get_row(table=table, md5=md5)
+            assert row
+            assert row.md5 == md5
+            assert row.min_processed_date == original_datetime
+            assert row.max_processed_date == original_datetime
+
+            # Update the row
+            new_datetime = "1999-01-01T23:59:59.1234567Z"
+            orm.update_row(table=table, md5=md5, min_date=new_datetime, max_date=new_datetime)
+            row = orm.get_row(table=table, md5=md5)
+            assert row.min_processed_date == new_datetime
+            assert row.max_processed_date == new_datetime
+
+        assert len(orm.get_all_rows(table=table)) == 3
+
+    # Try to obtain a non-existing item
+    assert orm.get_row(table=orm.Graph, md5="non-existing") is None
```

Comment:
This was considered a standalone test in the first place. We had `test_add and `test_get`. However, there is no point in a testing the `get` function without having previously invoked `add`, to add data to the database as the current implementation creates an empty DB. This is why it was decided to simplify the design and combine both tests as one.

Reply:
I'm thinking that you could parametrize the test to remove that last line. The parameters could be the hash, the date and the expected result of the row, this way that last case could be included in the 'regular' cases, but since the result wouldn't change a lot I'm OK with leaving this as it is.